### DISCUSSION TITLE
순찰 AI

### DISCUSSION THREAD

**Discussion by yj098** [naverz/zepeto-studio-kor#1058] [5. Q&A]
안녕하세요 혹시 타입스크립트에서는 순찰하는 AI는 어떻게 구현하나요??
유튜브에 c#으로 내비게이션 이용해서 만들 수 있는 강좌는 있는데 타입스크립트 강좌는 없네요 ㅠㅠ
혹시 이것과 관련해서 어떤 강좌를 보면 좋을지 추천 부탁 드립니다 ㅠOㅠ

**Comment by Zepeto-TechEvangelist** [comment]
안녕하세요, 맵에 NavMesh를 설정하고 AI에게 Nav Mesh Agent를 세팅하는 과정은 유니티와 동일하게 하시면 됩니다.

<img width="1110" alt="스크린샷 2022-07-26 오전 8 29 56" src="https://user-images.githubusercontent.com/100118650/180891743-64be8c28-7e20-4d7b-9021-876bc694ace8.png">

<img width="1110" alt="스크린샷 2022-07-26 오전 8 30 11" src="https://user-images.githubusercontent.com/100118650/180891766-712f8048-6398-4c16-9da6-04cda993a18f.png">


타겟을 찾아 움직이는 샘플 코드를 이동시킬 AI 오브젝트에 적용해 보세요.

```typescript
import { ZepetoScriptBehaviour } from 'ZEPETO.Script'
import { NavMesh, NavMeshAgent } from 'UnityEngine.AI';
import { Transform } from 'UnityEngine';

export default class AI extends ZepetoScriptBehaviour {
    
    private test : NavMesh;
    private agent : NavMeshAgent;
    public target : Transform;
     
    Awake() {
        this.agent = this.GetComponent<NavMeshAgent>();
    }
    
    Update(){

        this.agent.SetDestination(this.target.position)
    }

}
```



https://user-images.githubusercontent.com/100118650/180891905-3b463901-f9dd-438c-891c-680a7cc6226f.mov





성공하신 다른 분의 글도 참고 부탁 드립니다.
https://github.com/naverz/zepeto-studio-kor/discussions/483 
감사합니다.

**Reply by yj098** [reply]
귀한 시간 내주셔서 자세한 설명까지 감사합니다 ㅠㅠㅠ :)
